packages/opentelemetry-resources/src/platform/node/detectors/EnvDetector.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Resource, Labels } from '../../../Resource';
+
+/**
+ * EnvDetector can be used to detect the presence of and create a Resource
+ * from the OTEL_RESOURCE_LABELS environment variable.
+ */
+export class EnvDetector {
```

Comment:
I think specs is not yet defined for the env detector. Perhaps, we can open an issue in specs to start the discussion.

Reply:
Opened: https://github.com/open-telemetry/opentelemetry-specification/issues/535